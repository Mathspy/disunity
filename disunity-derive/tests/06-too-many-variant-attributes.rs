use disunity_derive::Variant;

#[derive(Variant)]
enum Class {
    #[disunity(discriminant = 4)]
    #[disunity(discriminant = 4)]
    GameObject { field: i32 },
}

fn main() {
    let _ = ClassVariant::GameObject;
}
