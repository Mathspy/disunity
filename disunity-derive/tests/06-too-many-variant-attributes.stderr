error: unexpected second #[disunity] attribute macro on variant
 --> tests/06-too-many-variant-attributes.rs:6:5
  |
6 |     #[disunity(discriminant = 4)]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared type `ClassVariant`
  --> tests/06-too-many-variant-attributes.rs:11:13
   |
11 |     let _ = ClassVariant::GameObject;
   |             ^^^^^^^^^^^^ use of undeclared type `ClassVariant`
